src/Microsoft.FeatureManagement/FeatureManagerSnapshot.cs

```diff
@@ -12,37 +12,99 @@ namespace Microsoft.FeatureManagement
     /// <summary>
     /// Provides a snapshot of feature state to ensure consistency across a given request.
     /// </summary>
-    class FeatureManagerSnapshot : IFeatureManagerSnapshot
+    class FeatureManagerSnapshot : IFeatureManagerSnapshot, IDynamicFeatureManagerSnapshot
     {
         private readonly IFeatureManager _featureManager;
+        private readonly IDynamicFeatureManager _dynamicFeatureManager;
         private readonly IDictionary<string, bool> _flagCache = new Dictionary<string, bool>();
-        private IEnumerable<string> _featureNames;
+        private readonly IDictionary<string, object> _variantCache = new Dictionary<string, object>();
+        private IEnumerable<string> _featureFlagNames;
+        private IEnumerable<string> _dynamicFeatureNames;
 
-        public FeatureManagerSnapshot(IFeatureManager featureManager)
+        public FeatureManagerSnapshot(
+            IFeatureManager featureManager,
+            IDynamicFeatureManager dynamicFeatureManager)
         {
             _featureManager = featureManager ?? throw new ArgumentNullException(nameof(featureManager));
+            _dynamicFeatureManager = dynamicFeatureManager ?? throw new ArgumentNullException(nameof(dynamicFeatureManager));
         }
 
-        public async IAsyncEnumerable<string> GetFeatureNamesAsync([EnumeratorCancellation]CancellationToken cancellationToken)
+        public async IAsyncEnumerable<string> GetFeatureFlagNamesAsync([EnumeratorCancellation]CancellationToken cancellationToken)
         {
-            if (_featureNames == null)
+            if (_featureFlagNames == null)
             {
                 var featureNames = new List<string>();
 
-                await foreach (string featureName in _featureManager.GetFeatureNamesAsync(cancellationToken).ConfigureAwait(false))
+                await foreach (string featureName in _featureManager.GetFeatureFlagNamesAsync(cancellationToken).ConfigureAwait(false))
                 {
                     featureNames.Add(featureName);
                 }
 
-                _featureNames = featureNames;
+                _featureFlagNames = featureNames;
             }
 
-            foreach (string featureName in _featureNames)
+            foreach (string featureName in _featureFlagNames)
             {
                 yield return featureName;
             }
         }
 
+        public async IAsyncEnumerable<string> GetDynamicFeatureNamesAsync([EnumeratorCancellation]CancellationToken cancellationToken = default)
```

Comment:
>cancellationToken = default [](http://example.com/codeflow?start=116&length=27)

default not needed here. Cancellation token already has a default value in the interface.